spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -372,7 +372,7 @@ public static void importSparkTable(SparkSession spark, TableIdentifier sourceTa
     try {
       PartitionSpec spec = SparkSchemaUtil.specForTable(spark, sourceTableIdentWithDB.unquotedString());
 
-      if (spec == PartitionSpec.unpartitioned()) {
+      if (Objects.equal(spec, PartitionSpec.unpartitioned())) {
```

Comment:
checking for ref. equality is probably fine here, but it takes a reader longer to navigate the code and figure out whether ref equality is really wanted here vs just using `equals()`